pkg/service/pipeline.go

```diff
@@ -322,6 +322,34 @@ func (s *service) UpdateNamespacePipelineByID(ctx context.Context, ns resource.N
 		return nil, err
 	}
 
+	toUpdTags := toUpdPipeline.GetTags()
+
+	// We will need to revert it after the frontend is ready.
+	if len(toUpdTags) > 0 {
+		currentTags := existingPipeline.TagNames()
+		toBeCreatedTagNames := []string{}
```

Comment:
would it make sense here to pre-allocate the slice with len(toUpdTags) for performance ?

Reply:
@tillknuesting 
Thanks for telling me it! I did not know it.
I researched a little. 
It seems it will be better if expected length is close to actual length for the memory & performance.

However, there are 2 slices here, which are `toBeCreatedTagNames` & `toBeDeletedTagNames`.
`len(toBeDeletedTagNames)` + `len(toBeCreatedTagNames)` will be equal to `len(toUpdTags)`
It makes it a trade-off here.

But, in this part, maybe there won't be too many tags in this part.
So, I guess two ways will be similar. 
**And, I will modify it by pre-allocation.**

Thank you for telling me this again! 
I will also consider it for future development